mmengine/structures/base_data_element.py

```diff
@@ -561,6 +562,16 @@ def to_dict(self) -> dict:
             for k, v in self.all_items()
         }
 
+    def pin_memory(self) -> 'BaseDataElement':
+        """BaseDataElement supports pin_memory function to speed up
+        training."""
+        for k, v in self.items():
+            if not isinstance(v, BaseDataElement):
```

Comment:
can simply replace with if hasattr(v, 'pin_memory') 
just follow the `cuda()` and `to()` method

Reply:
ok